controllers/configmap_controller.go

```diff
@@ -208,16 +214,32 @@ func (r *ConfigMapReconciler) deconfigureInstances(instances []*instance.Info, n
 	return nil
 }
 
-// hasAssociatedInstance returns true if the given node is associated with an instance in the given slice
-func hasAssociatedInstance(node *core.Node, instances []*instance.Info) bool {
-	for _, instanceInfo := range instances {
-		for _, nodeAddress := range node.Status.Addresses {
-			if instanceInfo.IPv4Address == nodeAddress.Address {
-				return true
+// hasAssociatedInstance returns true if the given node is associated with any instance in the given slice.
+// The instance's network address must be a valid IPv4 address or resolve to one.
+func hasAssociatedInstance(node *core.Node, instances []*instance.Info) (bool, error) {
```

Comment:
Given you are changing this method, I suggest passing just `node.Status.Addresses` given you are not using anything else in the node object.